src/Orchard.Web/Modules/Orchard.Localization/Services/LocalizationService.cs

```diff
@@ -87,5 +91,51 @@ public class LocalizationService : ILocalizationService {
             // Warning: May contain more than one localization of the same culture.
             return query.List().ToList();
         }
+
+        bool ILocalizationService.TryGetRouteForUrl(string url, out AutoroutePart route) {
+            route = _contentManager.Query<AutoroutePart, AutoroutePartRecord>()
+                .ForVersion(VersionOptions.Published)
+                .Where(r => r.DisplayAlias == url)
+                .List()
+                .FirstOrDefault();
+
+            if (route == null)
+                route = _homeAliasService.GetHomePage(VersionOptions.Latest).As<AutoroutePart>();
+            return route != null;
+        }
+
+        bool ILocalizationService.TryFindLocalizedRoute(ContentItem routableContent, string cultureName, out AutoroutePart localizedRoute) {
+            if (!routableContent.Parts.Any(p => p.Is<ILocalizableAspect>())) {
+                localizedRoute = null;
+                return false;
+            }
+
+            IEnumerable<LocalizationPart> localizations = ((ILocalizationService)this).GetLocalizations(routableContent, VersionOptions.Published);
+
+            ILocalizableAspect localizationPart = null, siteCultureLocalizationPart = null;
+            foreach (LocalizationPart l in localizations) {
+                if (l.Culture.Culture == cultureName) {
+                    localizationPart = l;
+                    break;
+                }
+                if (l.Culture == null && siteCultureLocalizationPart == null) {
+                    siteCultureLocalizationPart = l;
+                }
+            }
+
+            if (localizationPart == null) {
+                localizationPart = siteCultureLocalizationPart;
+            }
+
+            if (localizationPart == null) {
+                localizedRoute = null;
+                return false;
+            }
+
+            ContentItem localizedContentItem = localizationPart.ContentItem;
+            localizedRoute = localizedContentItem.Parts.Single(p => p is AutoroutePart).As<AutoroutePart>();
```

Comment:
Single is evil, it will throw if it's not exactly one element. But you might also want that.